File: android/core/src/main/kotlin/dev/example/moodlenative/core/SemesterSeason.kt
package dev.example.moodlenative.core

enum class SemesterSeason(
    val rawValue: String,
    val title: String,
    val sortOrder: Int,
) {
    SPRING("spring", "春", 0),
    FALL("fall", "秋", 1),
    UNKNOWN("unknown", "その他", 2),
}

data class AcademicSemester(
    val academicYear: Int,
    val season: SemesterSeason,
) {
    val id: String
        get() = "$academicYear-${season.rawValue}"

    val displayName: String
        get() = "${academicYear}年度${season.title}セメスター"

    val compactDisplayName: String
        get() = "${academicYear % 100} 年度${season.title}セメスター"
}

File: ios/Shared/Sources/Core/Models/SemesterSeason.swift
import Foundation

public enum SemesterSeason: String, Codable, Hashable, Sendable {
    case spring
    case fall
    case unknown

    public var title: String {
        switch self {
        case .spring:
            return "春"
        case .fall:
            return "秋"
        case .unknown:
            return "その他"
        }
    }

    public var sortOrder: Int {
        switch self {
        case .spring:
            return 0
        case .fall:
            return 1
        case .unknown:
            return 2
        }
    }
}
